core/domain/suggestion_registry.py

```diff
@@ -319,30 +318,124 @@ def accept(self, commit_message):
+            raise utils.ValidationError('Expected change_cmd to contain cmd')
+
+        if (
+                self.change_cmd.cmd !=
+                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION):
+            raise utils.ValidationError('Expected cmd to be %s, obtained %s' % (
+                question_domain.CMD_ADD_QUESTION_FROM_SUGGESTION,
+                self.change_cmd.cmd))
+
+        if not self.change_cmd.question_dict:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain question_dict')
+
+        if 'question_state_data' not in self.change_cmd.question_dict:
+            raise utils.ValidationError(
+                'Expected question dict to contain question_state_data')
+
+        if 'language_code' not in self.change_cmd.question_dict:
+            raise utils.ValidationError(
+                'Expected question dict to contain language_code')
+
+        if (
+                'question_state_schema_version' not in
+                self.change_cmd.question_dict):
+            raise utils.ValidationError(
+                'Expected question dict to contain'
+                ' question_state_schema_version')
+
+        if not self.change_cmd.skill_id:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain skill_id')
+
+    def pre_accept_validate(self):
+        """Performs referential validation. This function needs to be called
+        before accepting the suggestion.
         """
-        suggestion = cls(
-            suggestion_dict['suggestion_id'],
-            suggestion_dict['target_id'],
-            suggestion_dict['target_version_at_submission'],
-            suggestion_dict['status'], suggestion_dict['author_id'],
-            suggestion_dict['final_reviewer_id'], suggestion_dict['change_cmd'],
-            suggestion_dict['score_category'], suggestion_dict['last_updated'])
+        question_dict = self.change_cmd.question_dict
+        self.validate()
+        if (
+                question_dict['question_state_schema_version'] !=
+                feconf.CURRENT_EXPLORATION_STATES_SCHEMA_VERSION):
+            raise utils.ValidationError(
+                'Question state schema version is not up to date.')
+
+        skill_domain.Skill.require_valid_skill_id(self.change_cmd.skill_id)
+        skill = skill_services.get_skill_by_id(
+            self.change_cmd.skill_id, strict=False)
+        if skill is None:
+            raise utils.ValidationError(
+                'The skill with the given id doesn\'t exist.')
 
-        return suggestion
+
+    def get_change_list_for_accepting_suggestion(self):
+        pass
+
+    def accept(self, unused_commit_message):
+        """Accepts the suggestion.
+
+        Args:
+            unused_commit_message: str. This parameter is passed in for
+                consistency with the existing suggestions. As there is no model
+                being commited to, this arg is unused.
```

Comment:
spelling: committed

Also, I thought the reason for omitting it is because the commit message is autogenerated (please check)? I thought that every commit has a commit message.

Reply:
The commit message is actually assigned by default in the add_question function call. There is no need to pass it from this function. So this parameter is unused.

Reply:
OK, then say that instead. I think the existing docstring is a bit misleading.

Reply:
I had edited the comment accordingly in the previous commit.